subprojects/dependency-management/src/main/java/org/gradle/api/internal/artifacts/ivyservice/resolveengine/ProjectDependencyForcingResolver.java

```diff
@@ -0,0 +1,62 @@
+/*
+ * Copyright 2016 the original author or authors.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *      http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package org.gradle.api.internal.artifacts.ivyservice.resolveengine;
+
+import org.gradle.api.artifacts.Module;
+import org.gradle.api.artifacts.ModuleVersionIdentifier;
+import org.gradle.api.artifacts.component.ProjectComponentIdentifier;
+import org.gradle.internal.component.model.ComponentResolveMetaData;
+
+import java.util.ArrayList;
+import java.util.Collection;
+
+class ProjectDependencyForcingResolver implements ModuleConflictResolver {
+    private Module callingModule;
+    private final ModuleConflictResolver delegate;
+
+    public ProjectDependencyForcingResolver(Module callingModule, ModuleConflictResolver delegate) {
+        this.callingModule = callingModule;
+        this.delegate = delegate;
+    }
+    @Override
+    public <T extends ComponentResolutionState> T select(Collection<? extends T> candidates) {
+        // the collection will only be initialized if more than one project candidate is found
+        Collection<T> projectCandidates = null;
+        T foundProjectCandidate = null;
+        for(T candidate: candidates) {
+            ComponentResolveMetaData metaData = candidate.getMetaData();
+            if (metaData!=null && metaData.getComponentId() instanceof ProjectComponentIdentifier) {
+                ModuleVersionIdentifier moduleId = metaData.getId();
+                if (callingModule!=null && (callingModule.getGroup().equals(moduleId.getGroup())
+                && callingModule.getName().equals(moduleId.getName()))) continue;
+                if (foundProjectCandidate==null) {
+                    foundProjectCandidate = candidate;
+                } else {
+                    if (projectCandidates==null) {
+                        projectCandidates = new ArrayList<T>();
+                        projectCandidates.add(foundProjectCandidate);
+                    }
+                    projectCandidates.add(candidate);
+                }
+            }
+        }
+        if (projectCandidates!=null) {
+            return delegate.select(projectCandidates);
+        }
```

Comment:
What _exactly_ is the logic in this class doing? I'm not clear why we are excluding the current project from the candidates. Some clarifying comments and refactoring might help.